traitsui/tests/editors/test_text_editor.py

```diff
@@ -99,3 +156,85 @@ def test_text_editor_custom_style_placeholder(self):
+        # Smoke test to test setup and tear down of an editor.
+        # This test can be pull out to become toolkit-agnostic once
+        # wx TextEditor also implements dispose.
+        foo = Foo()
+        view = get_view(style=style, auto_set=auto_set)
+        with create_ui(foo, dict(view=view)):
+            pass
+
+    def test_simple_editor_init_and_dispose(self):
+        # Smoke test to test setup and tear down of an editor.
+        self.check_editor_init_and_dispose(style="simple", auto_set=True)
+
+    def test_custom_editor_init_and_dispose(self):
+        # Smoke test to test setup and tear down of an editor.
+        self.check_editor_init_and_dispose(style="custom", auto_set=True)
+
+    def test_simple_editor_init_and_dispose_no_auto_set(self):
+        # Smoke test to test setup and tear down of an editor.
+        self.check_editor_init_and_dispose(style="simple", auto_set=False)
+
+    def test_custom_editor_init_and_dispose_no_auto_set(self):
+        # Smoke test to test setup and tear down of an editor.
+        self.check_editor_init_and_dispose(style="custom", auto_set=False)
+
+    def test_simple_auto_set_update_text(self):
+        foo = Foo()
+        view = get_view(style="simple", auto_set=True)
+        gui = GUI()
+        with create_ui(foo, dict(view=view)) as ui:
+            editor, = ui.get_editors("name")
+            set_text(editor, "NEW")
+            gui.process_events()
+
+            self.assertEqual(foo.name, "NEW")
+
+    def test_simple_auto_set_false_do_not_update(self):
+        foo = Foo(name="")
+        view = get_view(style="simple", auto_set=False)
+        gui = GUI()
+        with create_ui(foo, dict(view=view)) as ui:
+            editor, = ui.get_editors("name")
+
+            set_text(editor, "NEW")
+            gui.process_events()
+
+            self.assertEqual(foo.name, "")
+
+            key_press_return(editor)
+            gui.process_events()
+
+            self.assertEqual(foo.name, "NEW")
+
+    def test_custom_auto_set_true_update_text(self):
+        # the auto_set flag is disregard for custom editor.
+        foo = Foo()
+        view = get_view(auto_set=True, style="custom")
+        gui = GUI()
+        with create_ui(foo, dict(view=view)) as ui:
+            editor, = ui.get_editors("name")
+
+            set_text(editor, "NEW")
+            gui.process_events()
+
+            self.assertEqual(foo.name, "NEW")
+
+    def test_custom_auto_set_false_update_text(self):
+        # the auto_set flag is disregard for custom editor.
+        foo = Foo()
+        view = get_view(auto_set=False, style="custom")
+        gui = GUI()
+        with create_ui(foo, dict(view=view)) as ui:
+            editor, = ui.get_editors("name")
+
+            set_text(editor, "NEW")
+            gui.process_events()
+
+            key_press_return(editor)
+            gui.process_events()
+
+            self.assertEqual(foo.name, "NEW\n")
```

Comment:
When https://github.com/enthought/traitsui/issues/752 is also fixed, these tests can be pulled out of the `skip_if_not_qt4` decorator and be run against wx. In the mean time, doing so may introduce more test interactions.

Reply:
Actually it might be easier if I pull these tests out now and skip on wx. Later when we get to #752, it should only require removing the skip and implementing the simulation functions (e.g. `key_press_return`) for wx.